File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int , int> mp;
        for(int x : nums){
            mp[x]++;
            if(mp[x] > nums.size()/2){
                return x;
            }
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        List<Integer> ans = new ArrayList<>();
        HashMap<Integer,Integer> mp = new HashMap<>();
        for(int i = 0; i < n; i++){
            mp.put(nums[i], mp.getOrDefault(nums[i], 0) + 1);
        }
        for(Map.Entry<Integer,Integer> en : mp.entrySet()){
            if(en.getValue() > n/2){
                return en.getKey();
            }
        }
        return -1;
    }
}